k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -860,6 +861,85 @@ public void fetchPart(Message message, Part part, MessageRetrievalListener<Messa
         }
     }
 
+    @Override
+    public void fetchPartWithProgressCallback(Message message, Part part, MessageRetrievalListener<Message> listener, AttachmentProgressCallback progressCallback) throws MessagingException {
```

Comment:
Yes, I think that is possible. I only need to pass the callback, so I can add a check for the `progressCallback` being null in the same code. Will do it shortly.